Assets/Mirror/Runtime/NetworkServer.cs

```diff
@@ -780,10 +784,31 @@ static void SpawnObserversForConnection(NetworkConnection conn)
                 {
                     // Debug.Log("Sending spawn message for current server objects name='" + identity.name + "' netId=" + identity.netId + " sceneId=" + identity.sceneId.ToString("X"));
 
-                    bool visible = identity.OnCheckObserver(conn);
-                    if (visible)
+                    // not force hidden?
+                    if (!identity.forceHidden)
                     {
-                        identity.AddObserver(conn);
+                        // legacy system support (for now)
+#pragma warning disable 618
+                        if (identity.visibility != null)
+                        {
+                            if (identity.visibility.OnCheckObserver(conn))
+                            {
+                                identity.AddObserver(conn);
+                            }
+                        }
```

Comment:
```suggestion
                        if (identity.visibility?.OnCheckObserver(conn))
                            identity.AddObserver(conn);
```
if the ? isn't valid here, at least combine the if's

Reply:
?. isn't because unity.
considered combining the ifs but this way it's more obvious since the AOI case below is the same